packages/components/src/mdx-components.tsx

```diff
@@ -0,0 +1,50 @@
+import fs from 'node:fs/promises';
+import path from 'node:path';
+import clsx from 'clsx';
+import { useMDXComponents as getDocsMDXComponents } from 'nextra-theme-docs';
+
+const docsComponents = getDocsMDXComponents({
+  async source({ src, type, ...props }) {
+    if (!src) {
+      throw new Error('Must provide `src` prop');
+    }
+    if (src.startsWith('/')) {
+      const filePath = path.join(process.cwd(), 'public', src);
+      try {
+        await fs.access(filePath);
+      } catch (error) {
+        const relativePath = path.relative(process.cwd(), filePath);
+        if ((error as any).code === 'ENOENT') {
+          throw new Error(`File doesn't exist: ${relativePath}`);
+        }
+        throw new Error(`Error checking file: ${relativePath}`);
+      }
+    }
```

Comment:
since I removed `next-videos` where we statically imported video, now I added a check of the existing file thanks to server components

Reply:
nice